math/src/main/java/eu/itesla_project/math/matrix/PlainMatrix.java

```diff
@@ -80,6 +103,25 @@ public PlainMatrix toPlain() {
     }
 
     @Override
+    public SparseMatrix toSparse() {
+        return (SparseMatrix) to(new SparseMatrixFactory());
+    }
+
+    @Override
+    public Matrix to(MatrixFactory factory) {
+        Objects.requireNonNull(factory);
+        if (factory instanceof PlainMatrixFactory) {
+            return this;
```

Comment:
This is what copy method does. to method just returns itself if same matrix impl is asked else returns a copy